packages/replayer/src/replay.utils.ts

```diff
@@ -130,31 +130,84 @@ const installVirtualEventLoop = (page: Page, sessionStartTime: DateTime) => {
   `);
 };
 
-export const getSessionStartTime = (sessionData: SessionData): DateTime =>
-  getMinMaxRrwebTimestamps(sessionData)[0];
+export const getSessionStartTime = (sessionData: SessionData): DateTime => {
+  const rrWebTimeRange = getMinMaxRrwebTimestamps(sessionData);
+  if (rrWebTimeRange != null) {
+    return rrWebTimeRange.min;
+  }
+
+  const userEventEventLog = sessionData.userEvents?.event_log ?? [];
+  if (userEventEventLog.length === 0) {
+    const logger = log.getLogger(METICULOUS_LOGGER_NAME);
+    logger.warn(
+      "No user or rrweb events, so cannot accurately determine start timestamp. Using current Date instead, however this will be unstable"
+    );
+    return DateTime.now().toUTC();
+  }
+
+  // event.timeStamp differs from event.timeStampRaw in that (a) it is relative to the start of the session,
+  // (b) it has been adjusted to subtract the time between constructing the recorder and calling.start().
+  //
+  // By subtracting event.timeStamp from event.timeStampRaw we get the performance.timeOrigin plus the
+  // the time between constructing the recorder and calling.start(), in other words: the time at which recording
+  // started.
```

Comment:
I would move this explainer to jsdoc on `timeStamp` and `timeStampRaw` fields.

Reply:
Yep that's a good call. Most of the bottom half will remaining, but the top half is much better suited to be on the fields. That will depend on https://github.com/alwaysmeticulous/meticulous-sdk/pull/168 merging I think.

Reply:
Let's get this one merged, and then can clean up as part of or after 168.

Reply:
:+1: alright